dali/kernels/signal/moving_mean_square.cc

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/kernels/signal/moving_mean_square.h"
+
+namespace dali {
+namespace kernels {
+namespace signal {
+
+namespace {
+
+/**
+ * Accurate type is such type, that doesn't require
+ * reset interval for maintaining numeric precision
+ */
+template<typename T>
+struct is_accurate {
+  static constexpr bool value = std::is_integral<T>::value && sizeof(T) <= 2;
+};
```

Comment:
The name doesn't tell much - it depends on reader refering to the comment to know what it is about.
I'd move this boolean to accumulator_type or just use is_integral _on the accumulator_ when this information is needed and forego this trait entirely.